---
output: 
  html_document:
    code_download: false
    toc: true                  # table of content true
    toc_depth: 3               # upto three depths of headings (specified by #, ## and ###)
    toc_float: true
    number_sections: true      # if you want number sections at each table header
    theme: united              # many options for theme, this one is my favorite.
    highlight: tango           # specifies the syntax highlighting style
    css: 'style.css'
params:
  manhattan:  "covid_1_manhattan.png"
  gwascat: "gwascat_subset.csv"
  output_file: "multiqc_report.html"
title: "`r paste0('Genomics England GWAS Report' , '') `"
author: ""
date: ""
---



<img src="logo.png" width=100 style="position:absolute;top:4px;bottom:4px;right:4px;" />


```{r include = FALSE}
knitr::opts_chunk$set(echo=FALSE)
```

```{r echo = FALSE}
source("DTable.R")
source("sanitise.R")

suppressPackageStartupMessages({

library(magrittr)
library(DT)
library(data.table)
  
library(htmltools)
library(stringr)
library(dplyr)
library(kableExtra)
   })
```

# Visualisations

Plots to explore GWAS results

```{r}
figure_number  <- "Figure 1: " 
figure_caption <- "Manhattan plot"
knitr::include_graphics("covid_1_manhattan.png", )
htmltools::tags$figcaption( style = 'caption-side: bottom; text-align: center; font-size: 85%%; color: #71879d',
                            htmltools::em(figure_number), 
                            htmltools::em(figure_caption))
```

```{r}
figure_number  <- "Figure 2: " 
figure_caption <- "QQ plot"
knitr::include_graphics("covid_1_qqplot_ci.png", )
htmltools::tags$figcaption( style = 'caption-side: bottom; text-align: center; font-size: 85%%; color: #71879d',
                            htmltools::em(figure_number), 
                            htmltools::em(figure_caption))
```

# GWAS catalogue information

```{r}
table   <- as.data.frame(data.table::fread("gwascat_subset.csv"))
toKeep <- c("SNPS", "MAPPED_TRAIT_URI", "INITIAL.SAMPLE.SIZE", "RISK.ALLELE.FREQUENCY", "DISEASE.TRAIT")
toKeep <- colnames(table)
DT::datatable(table[,toKeep],
               # OPTIONS:
                options = list(
                  filter     = "bottom",
                  # Does not allow columnful dataframes go rogue and tucks them in to fit page width
                  scrollX = TRUE,
                  scrollCollapse = TRUE,

                  # Defines all capabilities
                  dom        = 'BRrltpi',

                  autoWidth  = TRUE,
                  ColReorder = TRUE,
                  buttons    = list('copy','print', list(extend  = 'collection',
                                                         buttons = c('csv', 'excel', 'pdf'),
                                                         text    = 'Download'),
                                    I('colvis'))))


```
